include/swift/SIL/SILBuilder.h

```diff
@@ -2058,53 +2058,56 @@ class SILBuilder {
   }
 };
 
-/// An wrapper on top of SILBuilder's constructor that automatically sets the
-/// current SILDebugScope based on the specified insertion point. This is useful
-/// for situations where a single SIL instruction is lowered into a sequence of
-/// SIL instructions.
-class SILBuilderWithScope : public SILBuilder {
+/// A wrapper on top of SILBuilder's constructor that sets the insertion point,
+/// and automatically sets the SILDebugScope based on the specified IP. This is
+/// useful for situations where a single SIL instruction is lowered into a
+/// sequence of SIL instructions.
+class SILBuilderForCodeExpansion : public SILBuilder {
   void inheritScopeFrom(SILInstruction *I) {
     assert(I->getDebugScope() && "instruction has no debug scope");
     setCurrentDebugScope(I->getDebugScope());
   }
 
 public:
-  explicit SILBuilderWithScope(
+  explicit SILBuilderForCodeExpansion(
```

Comment:
Would be nice to add a doxygen comment for the constructor that explains that InsertedInstrs returns a list of the instruction built by this builder etc.